src/app/frontend/common/namespace/module.js

```diff
@@ -42,22 +42,21 @@ export default angular
  */
 function ensureNamespaceParamPresent($location, $transitions, $state) {
   /**
-   * Helper function which replaces namespace URL search param when the given namespace is
-   * undefined.
-   * @param {string|undefined} namespace
+   * Helper function which redirect namespace param when the given namespace
+   * in the transition is undefined.
    * @returns {boolean}
    */
-  function replaceUrlIfNeeded(namespace) {
-    if (namespace === undefined && !!$state.transition &&
-        $state.transition.to().name !== loginState) {
-      $location.search(namespaceParam, DEFAULT_NAMESPACE);
-      $location.replace();
-      return false;
```

Comment:
This line causes the bug since it will abandon the transition.
Also, previously the namespace is set by watching location.search, but a better solution could be redirecting the state by checking if the namespace is giving.